### DISCUSSION TITLE
Battery light not coming on

### DISCUSSION THREAD

**Discussion by mfletch3141** [gizmo-platform/gizmo#41] [Q&A] [answered]
I have 2 Gizmo models: v1.00 and v00.r4B
In both, I have a battery connected using the switch assembly and a fuse. Reading consistently 7.6V and higher. All status LED's are on. The Network and Field lights are blinking red rapidly. I have downloaded and copied the latest firmware onto the driver station and the system Pi. My DS is a Pi Zero 2 W. My system processor is a Pico WH and my student processor is a Pico H. 
I have uploaded the firmware appropriate to each gizmo version. No matter the permutation of steps, the battery light is consistently off.
The only change I have observed is that in one permutation, the field and network lights turned purple, but never went beyond that.

I have watched all of the videos in the BESTedu site, read the documentation on the github site, and read the two pdf files on the workflow. I have also searched this forum for references to the battery light not coming on. 

Please advise. Thank you!
Mike Fletcher

**Comment by mfletch3141** [comment]
Follow up:
After finding the troubleshooting guide, I was able to see the output of the DS and the Gizmo. 
![connected to DS and not configuring](https://github.com/user-attachments/assets/be8b18f9-a424-4bab-8c12-dd187287d6fa)
![DS initial boot](https://github.com/user-attachments/assets/1ff031f1-4b13-46d1-8a82-2a18bdf7a7aa)
![DS verifying connect to gizmo hardware](https://github.com/user-attachments/assets/9eb49356-93ba-4a19-b1e2-628c90dd5479)

The only variable is the gssconfig file that is team specific. There is no reference to that file in the BEST edu directions. I copied it to the DS root directory with the rest of the ramdisk files. The picture in the directions on the PPT had other files on it, so I wasn't sure I put it in the right place. Please advise.

**Reply by barulicm** [reply, marked as answer]
The gssconfig file is no longer used in the ramdisk version of the DS software. The team assignment is handled with the name of the SD card drive. The other values that used to come from the gssconfig file are now generated by the DS. (I think the PPTs you're looking at may be from before the newer releases changed this.)

Blinking red network and field lights indicates your Gizmo is not bound to any driver station. This is consistent with the logs from your gizmo. `GIZMO_CONFIG_WAIT` is the log message sent when the Gizmo is waiting for a new config to be sent over serial (USB from the DS). The battery light doesn't turn on until binding is finished. (That's an error in the troubleshooting graphic. @the-maldridge, it might be nice to turn that light on during config loading.)

I'm assuming you have tried the binding process ( [bestedu](https://bestedu.bestrobotics.org/courses/gizmo-setup-for-hubs/lessons/how-to-bind-the-gizmo-to-a-driver-station/), [Gizmo docs](https://gizmoplatform.dev/startup/firmware.html#bind-the-gizmo-to-a-drivers-station) ). Did you leave the Gizmo connected to the DS during binding long enough for it to restart and connect?

Purple on the network and field lights means there was an error parsing the config on the Gizmo. That might indicate a corrupt configuration on the Gizmo. You can wipe the system processor entirely to make sure you're starting with a clean slate by flashing [the "nuke firmware" ](https://cdn-learn.adafruit.com/assets/assets/000/099/419/original/flash_nuke.uf2?1613329170). After that firmware boots up and clears the internal memory, you can reinstall the Gizmo system firmware and attempt binding again.

**Reply by PhotonBandit** [reply]
I had this same issue of what I thought was the battery light not coming on. Main issue was actually that I couldnt get DS to bind (as stated by @barulicm) but figured this might battery status LED might have something to do with it. Tried binding every which way once more before going ahead and reloading the DS SD card files to the latest version. Couldnt tell which version was on the SD card... files looked to be from last year (Fall 2024). Loading the much newer V0.2.0 did the trick. The DS binded right away and the battery light of course turned on. Good luck.
